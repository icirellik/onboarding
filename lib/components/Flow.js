import React from 'react';
import PropTypes from 'prop-types';
import SectionList from './SectionList';

/**
 * A flow contains mutliple different sections that each have completable
 * skills.
 */
class Flow extends React.PureComponent {
  render() {
    const { title, sections } = this.props.flow.data[0];
    return (
      <div>
        <h3>{title}</h3>
        <SectionList sections={sections} />
      </div>
    );
  }
}

Flow.propTypes = {
  flow: PropTypes.shape({
    title: PropTypes.string.isRequired,
    data: PropTypes.array.isRequired,
  }).isRequired,
};

export default Flow;
