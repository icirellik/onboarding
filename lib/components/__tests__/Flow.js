import React from 'react';
import Flow from '../Flow';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Flow', () => {

  const testProps = {
    flow: {
      title: "title",
      data: [{
        sections: [],
      }],
    },
  };

  it('renders correctly', () => {

    const wrapper = shallow(
      <Flow
        {...testProps}
      />
    );

    expect(wrapper.find('SectionList').length).toBe(1);
    expect(wrapper).toMatchSnapshot();

  });

});
