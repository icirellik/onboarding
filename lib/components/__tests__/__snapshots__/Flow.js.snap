// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flow renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Flow
    flow={
      Object {
        "data": Array [
          Object {
            "sections": Array [],
          },
        ],
        "title": "title",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3 />,
        <SectionList
          sections={Array []}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "sections": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3 />,
          <SectionList
            sections={Array []}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "sections": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
